### PR TITLE
Automatically install missing packages

### PR DESCRIPTION
**Pull request by jcohenadad**
This is a branch off of https://github.com/neuropoly/slicer-manual-annotation/pull/41

TODO:

- [x] Remove installation issue from [README troubleshooting](https://github.com/neuropoly/slicer-manual-annotation?tab=readme-ov-file#trouble-shooting)

TODO in subsequent PRs:

* Check if package installation failed --> https://github.com/neuropoly/slicer-manual-annotation/issues/44

### CODE REVIEW

File: SlicerCART/src/SlicerCART.py [hunk 1]
```diff
@@ -1546,11 +1619,12 @@ def cast_segmentation_to_uint8(self):
...
               segm = nib.load(case)
-              segm_data = segm.get_fdata()
+              segm_data_dtype = segm.dataobj.dtype
               print(f'infile: {input_path}, dtype: {segm_data.dtype}')
-              if segm_data.dtype != np.uint8:
+              if segm_data_dtype != np.uint8:
                   segm_data = segm_data.astype(np.uint8)
-                  segm_nii = nib.Nifti1Image(segm_data, segm.affine)
+                  segm.header.set_data_dtype(np.uint8)
+                  segm_nii = nib.Nifti1Image(segm_data, segm.affine, segm.header)
```
Comment (jcohenadad) [inline SlicerCART/src/SlicerCART.py:1627]: These changes seem unrelated to package management. For the sake of clarity in the git history, should we do avoid unrelated changes in this PR, and address these changes in another, more specific, PR?

File: SlicerCART/src/general_config.yml [hunk 2]
```diff
@@ -3,13 +3,16 @@ input_filetype: '*.nii.gz' # '*.nrrd' or '*.nii.gz'
...
 
-impose_bids_format: True
+impose_bids_format: False
 
 is_classification_requested: True
 is_segmentation_requested: True
 is_semi_automatic_phe_tool_requested: True
+
+ct_window_width: 85
+ct_window_level: 45
```
Comment (jcohenadad) [inline SlicerCART/src/general_config.yml:18]: Also unrelated to package management

**Comment by jcohenadad** [comment]
Tested on fresh Slicer install, and issue related to the missing qt import:

```console
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Applications/Slicer.app/Contents/lib/Python/lib/python3.9/imp.py", line 169, in load_source
    module = _exec(spec, sys.modules[name])
  File "<frozen importlib._bootstrap>", line 613, in _exec
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/julien/code/slicer-manual-annotation/SlicerCART/src/SlicerCART.py", line 67, in <module>
    check_and_install_python_packages()
  File "/Users/julien/code/slicer-manual-annotation/SlicerCART/src/SlicerCART.py", line 52, in check_and_install_python_packages
    response = qt.QMessageBox.question(slicer.util.mainWindow(), 'Install Extensions', msg,
NameError: name 'qt' is not defined
[Qt] loadSourceAsModule - Failed to load file "/Users/julien/code/slicer-manual-annotation/SlicerCART/src/SlicerCART.py"  as module "SlicerCART" !
[Qt] Fail to instantiate module  "SlicerCART"
[Qt] The following modules failed to be instantiated:
[Qt]    SlicerCART
[Qt] Populating font family aliases took 106 ms. Replace uses of missing font family ".AppleSystemUIFont" with one that exists to avoid this cost. 
```

**Comment by jcohenadad** [comment]
Example of qt box on Slicer fresh install:

image

File: SlicerCART/src/SlicerCART.py (hunk 1 above)
Comment (laurentletg) [reply SlicerCART/src/SlicerCART.py:1629]: Yes, sorry, I should have created another issue and PR specific to this, a small bug I found while testing the module. Note: `.get_fdata()` returns a floating point image so `if segm_data.dtype != np.uint8:` would always have returned `False`.
